urllib3/connection.py

```diff
@@ -265,7 +265,14 @@ def connect(self):
                     'for details.)'.format(hostname)),
                     SubjectAltNameWarning
                 )
-            match_hostname(cert, self.assert_hostname or hostname)
+
+            # In case the hostname is an IPv6 address, strip the square
+            # brackets from it before using it to validate. Non IPv6 addresses
+            # should not start/end with square brackets, so this should be
+            # safe.
```

Comment:
I don't mean explaining what it's doing but moreso why. Worth saying that this is something that match_hostname specifically demands?